asto-core/src/main/java/com/artipie/asto/events/EventQueue.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2023 artipie.com
+ * https://github.com/artipie/asto/LICENSE.txt
+ */
+package com.artipie.asto.events;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+/**
+ * Events queue with {@link ConcurrentLinkedQueue} under the hood.
+ * Instance of this class can be passed where necessary (to the adapters for example)
+ * to add data for processing into the queue.
+ * @param <T> Queue item parameter type.
+ * @since 1.17
+ */
+public final class EventQueue<T> {
```

Comment:
Why do we need this wrapper? 
Can we use `ConcurrentLinkedQueue` directly?

Reply:
To hide the queue from being modified (cleaned or polled) from anywhere except this package. Only adding elements should be allowed from outside.